### PR TITLE
Fix some cyclo issues

### PR DESCRIPTION
**Pull request by owenrumney**
- add GetAttribute to do nested attributes

### CODE REVIEW

File: internal/app/tfsec/rules/aws004.go [hunk 1]
```diff
@@ -60,49 +58,57 @@ func init() {
...
+
+			protocolAttr := resourceBlock.GetAttribute("protocol")
+
+			if protocolAttr != nil {
+				if protocolAttr.IsResolvable() && protocolAttr.Equals("HTTPS") {
+					return
+				}
+				if protocolAttr.IsResolvable() && protocolAttr.Equals("HTTP") {
+					// check if this is a redirect to HTTPS - if it is, then no problem
+					if redirectProtocolAttr := resourceBlock.GetAttribute("default_action/redirect/protocol"); redirectProtocolAttr != nil {
```
Comment (owenrumney) [inline internal/app/tfsec/rules/aws004.go:74]: @liamg  - Interested to know your thoughts on this

File: internal/app/tfsec/block/hclblock.go [hunk 2]
```diff
@@ -208,6 +211,27 @@ func (block *HCLBlock) GetAttribute(name string) Attribute {
 	return nil
 }
 
+func (block *HCLBlock) GetNestedAttribute(name string) Attribute {
+	parts := strings.Split(name, "/")
+	blocks := parts[:len(parts)-1]
+	attrName := parts[len(parts)-1]
+
+	working := block
```
Comment (liamg) [inline internal/app/tfsec/block/hclblock.go:219]: If we cast it to the interface we can remove the cast on line 224 and it'll be future-proofed.
```suggestion
	working := block.(Block)
```

File: internal/app/tfsec/block/hclblock.go [hunk 3]
```diff
@@ -197,6 +197,9 @@ func (block *HCLBlock) GetAttributes() []Attribute {
 }
 
 func (block *HCLBlock) GetAttribute(name string) Attribute {
+	if strings.Contains(name, "/") {
```
Comment (liamg) [inline internal/app/tfsec/block/hclblock.go:200]: Can block names definitely not contain `/`? If they can we could just ask that people call `GetNestedAttribute` directly.

File: internal/app/tfsec/rules/aws004.go (hunk 1 above)
Comment (liamg) [reply internal/app/tfsec/rules/aws004.go:74]: Looks good, I like it :+1:

**Review by liamg — CHANGES_REQUESTED** [review]

File: internal/app/tfsec/block/hclblock.go (hunk 3 above)
Comment (owenrumney) [reply internal/app/tfsec/block/hclblock.go:200]: I don't think so, but I agree it should be an explicit action so have fixed

File: internal/app/tfsec/block/hclblock.go (hunk 2 above)
Comment (owenrumney) [reply internal/app/tfsec/block/hclblock.go:219]: Done

**Review by liamg — APPROVED** [review]

**Pull request merged**